extensions/amp-playbuzz/0.1/amp-playbuzz.js

```diff
@@ -0,0 +1,263 @@
+      });
+      this.getViewport().onChanged(
+        utils.debounce(this.sendScrollDataToItem_.bind(this), 250));
+    });
+  }
+
+  createPlaybuzzLoader_() {
+    const doc = this.element.ownerDocument;
+    const createElement = utils.getElementCreator(doc);
+
+    const loaderImage = createElement('img', 'pb_feed_anim_mask');
+    loaderImage.src = logo;
+
+    const loaderText = createElement('div', 'pb_feed_loading_text');
+    loaderText.textContent = 'Loading...';
+
+    const loadingPlaceholder =
+      createElement('div', 'pb_feed_placeholder_container',
+        createElement('div', 'pb_feed_placeholder_inner',
+          createElement('div', 'pb_feed_placeholder_content', [
+            createElement('div', 'pb_feed_placeholder_preloader', loaderImage),
+            createElement('div', 'pb_ie_fixer'),
+            loaderText,
+          ])));
+
+    return loadingPlaceholder;
+  }
+
+  /**
+   * @param {number} height
+   *
+   * @memberOf AmpPlaybuzz
+   */
+  setElementSize_(height) {
+
+    if (isNaN(height) || height === this.itemHeight_) {
+      return;
+    }
+
+    this.itemHeight_ = height; //Save new height
+
+    if (this.win.document.readyState !== 'complete') {
+      return;
+    }
+
+    const heightInPixels = this.itemHeight_ + 'px';
+    // this.changeHeight(this.itemHeight_); //Is that better than using vSync mutate + setStyles ?
+    this.getVsync().mutate(() => {
+      setStyles(this.element, {'height': heightInPixels, 'width': '100%'});
+    });
+
+    this.getVsync().mutate(() => {
+      setStyles(this.iframe_, {
+        'height': heightInPixels,
+        'max-height': heightInPixels,
+        'width': '100%',
+      });
+    });
+  }
+
+
+  listenToPlaybuzzItemMessage_(messageName, handler) {
+    events.listen(this.win, 'message',
+      event => utils.handleMessageByName(event, messageName, handler));
+  }
+
+  generateEmbedSourceUrl_() {
+    const params = {};
+    params.itemUrl = this.item_.replace('https://', '//').replace('http://', '//').split('#')[0];
+    params.relativeUrl = params.itemUrl.split('.playbuzz.com')[1];
+    params.displayItemInfo = this.displayItemInfo_;
+    params.displayShareBar = this.displayShareBar_;
+    params.displayComments = this.displayComments_;
+    params.windowUrl = this.win.location;
+    params.parentUrl = params.windowUrl.href.split(params.windowUrl.hash)[0];
+    params.parentHost = params.windowUrl.hostname;
+
+    const embedUrl = utils.composeEmbedUrl(params);
+    return embedUrl;
+  }
+
+  sendScrollDataToItem_(changeEvent) {
+    const viewport = this.getViewport();
+
+    const scrollingData = {
+      event: 'scroll',
+      windowHeight: changeEvent.height,
+      scroll: changeEvent.top,
+      offsetTop: viewport.getLayoutRect(this.element).top,
+    };
+
+    const data = JSON.stringify(scrollingData);
+    postMessage(this.iframe_, 'onMessage', data, '*', false);
+  }
+
+  /** @override */
+  unlayoutOnPause() {
```

Comment:
Do you ever have videos or other playing/animating content? Generally, if you return `false` here, we require that you send (and process) message to your embed to stop heavy CPU/bandwidth operations. If you don't have this message ready, let's start by returning `true` here and destroying the iframe in the `unlayoutCallback` and then we can revisit.